File: frontend/job-admin/src/api/job.ts
import axios from 'axios';

const API_BASE_URL = 'http://localhost:5002/api';

// 设置 axios 默认配置
axios.interceptors.request.use((config) => {
  const token = localStorage.getItem('token');
  if (token) {
    config.headers.Authorization = `Bearer ${token}`;
  }
  config.headers['Content-Type'] = 'application/json';
  config.withCredentials = true;  // 添加这个配置以支持跨域请求携带凭证
  return config;
});

export interface Job {
  id: number;
  job_title: string;
  company_name: string;
  location: string;
  responsibilities: string[];
  requirements: string[];
  raw_jd_text: string;
  created_at: string;
  updated_at: string;
}

export interface JobListResponse {
  status: string;
  jobs: {
    id: number;
    job_title: string;
    company_name: string;
    location: string;
    created_at: string;
  }[];
  pagination: {
    total: number;
    pages: number;
    current_page: number;
    per_page: number;
    has_next: boolean;
    has_prev: boolean;
  };
}

export interface CreateJobResponse {
  status: string;
  job_id: number;
  message: string;
  job: Job;
}

export interface JobRecommendation {
  job_id: number;
  match_score: number;
  match_analysis: string;
  advantages: string[];
  challenges: string[];
  suggestions: string[];
  job_details: Job;
}

export interface MatchJobsResponse {
  status: string;
  recommendations: JobRecommendation[];
  message?: string;
}

// 创建新职位
export const createJob = async (rawJdText: string): Promise<CreateJobResponse> => {
  try {
    const response = await axios.post(`${API_BASE_URL}/job/admin/create_job`, {
      raw_jd_text: rawJdText,
    });
    return response.data;
  } catch (error: any) {
    console.error('Create job error:', error.response || error);
    throw error;
  }
};

// 获取职位列表
export const getJobs = async (params: {
  job_title?: string;
  location?: string;
  page?: number;
  per_page?: number;
}): Promise<JobListResponse> => {
  const response = await axios.get(`${API_BASE_URL}/job`, { params });
  return response.data;
};

// 获取职位详情
export const getJobById = async (jobId: number): Promise<{ status: string; job: Job }> => {
  const response = await axios.get(`${API_BASE_URL}/job/${jobId}`);
  return response.data;
};

// 匹配职位
export const matchJobs = async (params: {
  desired_position: string;
  desired_location: string;
}): Promise<MatchJobsResponse> => {
  const response = await axios.get(`${API_BASE_URL}/job/match`, { params });
  return response.data;
};

File: app/routes/job.py
from flask import Blueprint, request, jsonify, current_app
from app import db
from app.models import Job
from app.middleware import admin_required, jwt_required
from app.services.ai_service import parse_job_description
from datetime import datetime, UTC
import json

bp = Blueprint('job', __name__, url_prefix='/job')

@bp.route('/admin/create_job', methods=['POST', 'OPTIONS'])
@admin_required
def create_job():
    # 明确处理 OPTIONS 请求
    if request.method == 'OPTIONS':
        response = jsonify({'status': 'ok'})
        response.headers['Access-Control-Allow-Origin'] = 'http://localhost:5177'
        response.headers['Access-Control-Allow-Methods'] = 'POST, OPTIONS'
        response.headers['Access-Control-Allow-Headers'] = 'Content-Type, Authorization'
        return response, 204
        
    data = request.get_json()
    if not data or 'raw_jd_text' not in data:
        return jsonify({'error': 'Missing raw_jd_text'}), 400

    try:
        # 使用 AI 服务解析 JD
        parsed_data = parse_job_description(data['raw_jd_text'])

        job = Job(
            job_title=parsed_data['job_title'],
            company_name=parsed_data['company_name'],
            location=parsed_data['location'],
            raw_jd_text=data['raw_jd_text']
        )
        job.set_responsibilities(parsed_data['responsibilities'])
        job.set_requirements(parsed_data['requirements'])

        db.session.add(job)
        db.session.commit()

        return jsonify({
            'status': 'ok',
            'job_id': job.id,
            'message': 'Job created successfully',
            'job': job.to_dict()
        }), 201

    except ValueError as e:
        current_app.logger.error(f"Value Error in create_job: {str(e)}")
        return jsonify({'error': str(e)}), 400
    except Exception as e:
        current_app.logger.error(f"Error creating job: {str(e)}")
        db.session.rollback()
        return jsonify({'error': 'Internal server error'}), 500

@bp.route('/', methods=['GET'])
def list_jobs():
    # Get query parameters
    job_title = request.args.get('job_title', '')
    location = request.args.get('location', '')
    page = request.args.get('page', 1, type=int)
    per_page = request.args.get('per_page', 10, type=int)

    # Build query
    query = Job.query
    if job_title:
        query = query.filter(Job.job_title.ilike(f'%{job_title}%'))
    if location:
        query = query.filter(Job.location.ilike(f'%{location}%'))

    # Execute query with pagination
    pagination = query.order_by(Job.created_at.desc()).paginate(
        page=page, per_page=per_page, error_out=False
    )
    
    return jsonify({
        'status': 'ok',
        'jobs': [
            {
                'id': job.id,
                'job_title': job.job_title,
                'company_name': job.company_name,
                'location': job.location,
                'created_at': job.created_at.isoformat()
            }
            for job in pagination.items
        ],
        'pagination': {
            'total': pagination.total,
            'pages': pagination.pages,
            'current_page': page,
            'per_page': per_page,
            'has_next': pagination.has_next,
            'has_prev': pagination.has_prev
        }
    })

@bp.route('/<int:job_id>', methods=['GET'])
def get_job(job_id):
    job = Job.query.get_or_404(job_id)
    return jsonify({
        'status': 'ok',
        'job': job.to_dict()
    })
